src/main/java/net/mcreator/blockly/java/blocks/TextJoinBlock.java

```diff
@@ -41,8 +41,21 @@
 				return;
 			}
 
+			boolean hasString = false;
+			for (Element element : elements) {
+				if (element != null && BlocklyToCode.directProcessOutputBlock(master, element).startsWith("\"")) {

```

Comment:
This could detect code such as `"test".substring(3, 2)` as a plain string

Reply:
The substring operation adds () around the output so it doesn't get detected

Reply:
This is true, but there may be plugins not adding () around, so I don't think we should rely on this

Reply:
I replaced this check with a regex check, it should only match uninterrupted strings now